Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.Model.swift

```
@@ -0,0 +1,388 @@
+            } else {
+                return nil
+            }
+        }
+        
+        /// Adds a graphic to the graphics overlay.
+        /// - Parameters:
+        ///   - location: The `Point` location to place the graphic..
+        ///   - traceLocationType: The textual description of the trace location type.
+        private func addGraphic(for location: Point, traceLocationType: String) {
+            parametersOverlay.addGraphic(
+                Graphic(
+                    geometry: location,
+                    attributes: ["TraceLocationType": traceLocationType],
+                    symbol: nil
+                )
+            )
+        }
+        
+        /// Sets the selected filter barrier category and updates the status text.
+        func selectCategory(_ category: UtilityCategory) {
+            selectedCategory = category
+            traceEnabled = true
+            statusText = "\(category.name) selected."
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        @MainActor
+        func trace() async throws {
+            tracingActivity = .runningTrace
+            
+            guard let configuration = makeTraceConfiguration(category: selectedCategory) else {
+                statusText = "Failed to get trace configuration."
+                tracingActivity = .none
+                resetEnabled = true
+                return
+            }
+            traceParameters.traceConfiguration = configuration
+            var traceResults = [UtilityElementTraceResult]()
+            do {
+                traceResults = try await utilityNetwork
+                    .trace(using: traceParameters)
+                    .compactMap { $0 as? UtilityElementTraceResult }
+            } catch {
+                statusText = "Trace failed."
+                traceEnabled = true
+                traceCompleted = false
```

for this to do what you think it's doing you'd want to return or rethrow. Returning would eat the error, which wouldn't be what you want I don't think.